test/blackbox-tests/test-cases/no-name-field/run.t

```diff
@@ -17,3 +17,24 @@ executable(s) stanza works the same way
   File "dune", line 1, characters 0-36:
   Error: name field may not be omitted before dune version 1.1
   [1]
+
+there's only a public name but it's invalid as a name
+
+  $ dune build --root public-name-invalid-name
+  File "dune", line 1, characters 22-28:
+  Error: invalid library name.
+  Hint: library names must be non-empty and composed only of the following characters: 'A'..'Z',  'a'..'z', '_'  or '0'..'9'.
+  Public library names don't have this restriction. You can either change this public name to be a valid library name or add a "name" field with a valid library name.
+  [1]
+
+there's only a public name which is invalid, but sine the library is unwrapped,
+it's just a warning
+
+  $ dune build --root public-name-invalid-wrapped-false
+  Info: creating file dune-project with this contents: (lang dune 1.1)
+  File "dune", line 3, characters 14-21:
+  Warning: invalid library name.
+  Hint: library names must be non-empty and composed only of the following characters: 'A'..'Z',  'a'..'z', '_'  or '0'..'9'.
+  This is temporary allowed for libraries with (wrapped false).
+  It will not be supported in the future. Please choose a valid name field.
+  Entering directory 'public-name-invalid-wrapped-false'
```

Comment:
Could you also add a test with:

```
(library
 (wrapped false)
 (name foo.bar))
```

? i.e. using `name` instead of `public_name`, just to be sure that this case is tested as well.